zip-0320.rst

```diff
@@ -5,7 +5,242 @@
   Owners: Daira Emma Hopwood <daira@electriccoin.co>
           Kris Nuttycombe <kris@nutty.land>
   Credits: Hanh
-  Status: Reserved
+  Status: Draft
   Category: Standards / Wallet
   Discussions-To: <https://github.com/zcash/zips/issues/757>
 
+Terminology
+===========
+
+The key words "MUST", "MUST NOT", "SHOULD", and "MAY" in this document are
+to be interpreted as described in BCP 14 [#BCP14]_ when, and only when, they
+appear in all capitals.
+
+The terms below are to be interpreted as follows:
+
+Abstract
+========
+
+This ZIP defines a new encoding for transparent Zcash addresses. Wallets must
+ensure that no shielded notes are spent in transactions that send to a
+transparent address encoded in the specified fashion.
+
+This ZIP is presently in Draft status, and defines two alternate encodings 
+
+Motivation
+==========
+
+The Binance cryptocurrency exchange requires that funds sent to their deposit
+addresses come from source addresses that are readily identifiable using
+on-chain information, such that if necessary funds may be rejected by sending
+them back to the source address(es). This ZIP is intended to standardize 
+a transparent address encoding that is only usable by wallets that understand
+and will respect this constraint.
+
+Requirements
+============
+
+1. An entity wishing to receive funds from exclusively transparent sources
+   must be able to generate a receiving address such that only transparent
+   funds will be spent in transactions sent to this address.
+2. Wallets that have not been upgraded to recognize the new address format
+   cannot mistake the address for another address type or inadvertently
+   send shielded funds the address.
+3. No changes to recipient infrastructure beyond changes to address encoding
+   and decoding should be required as a consequence of this ZIP.
+
+Specification (Alternative 1)
+=============================
+
+TEX Addresses
+-------------
+
+A TEX Address is a ``bech32m`` reencoding of a transparent Zcash P2PKH address
+[#protocol-taddr]_ 
+
+Motivations for Alternative 1
+-----------------------------
+
+The TEX Address is the simplest-possible approach to creating a new address
+type that indicates that only transparent sources of funds should be used.
+
+Detailed Specification (Alternative 1)
+--------------------------------------
+
+A TEX address may be produced from a mainnet Zcash P2PKH Address by executing the
+following steps:
+
+1. Decode the address to a byte array using the ``base58check`` decoding
+   algorithm.
+2. Remove the two-byte address prefix from the resulting byte array. These
+   bytes should be equal to :math:`\mathtt{[0x1C, 0xB8]}`. The remainder of the
+   byte array (the **validating key hash**) should have a length of 20 bytes.
```

Comment:
```suggestion
2. If the length of the resulting byte sequence is not 22 bytes or if its two-byte
   address prefix is not :math:`\mathtt{[0x1C, 0xB8]}`, return an error. Otherwise,
   let the **validating key hash** be the remaining 20 bytes of the array after
   removing the two-byte address prefix.
```